### PR TITLE
Update SEOPress added new version functions

### LINKED ISSUE #5839: Update SEOPress integration for sitemap preloading
https://github.com/wp-seopress/wp-rocket/tree/feat/seopress-sitemap-preloading

### PR DESCRIPTION
**Pull request by CrochetFeve0251**
## Description

Update SEOPress to not use anymore deprecated SEOPress functions.

Fixes #5839

## Type of change

Please delete options that are not relevant.

- [ ] Enhancement (non-breaking change which improves an existing functionality)

## Is the solution different from the one proposed during the grooming?

No.

## How Has This Been Tested?
- [ ] Automated Tests 

# Checklist:

Please delete the options that are not relevant.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: inc/ThirdParty/Plugins/SEO/SEOPress.php [hunk 1]
```diff
@@ -27,9 +27,14 @@ public function __construct( Options_Data $option ) {
 	 * Subscribed events.
 	 */
 	public static function get_subscribed_events() {
-		if ( ! function_exists( 'seopress_get_toggle_xml_sitemap_option' ) || 1 !== (int) seopress_get_toggle_xml_sitemap_option() ) {
+		if ( ! function_exists( 'seopress_get_toggle_option' ) || '1' !== seopress_get_toggle_option( 'xml-sitemap' ) ) {
```
Comment (Tabrisrp) [inline inc/ThirdParty/Plugins/SEO/SEOPress.php:30]: What happens for someone with an older version of the SEOPress and this version of WPR? Does this function exist in the older versions?

Comment (remyperona) [inline inc/ThirdParty/Plugins/SEO/SEOPress.php]: Are we sure the function correctly returns an int?

File: inc/ThirdParty/Plugins/SEO/SEOPress.php [hunk 2]
```diff
@@ -27,9 +27,14 @@ public function __construct( Options_Data $option ) {
...
 			return [];
 		}
+
+		if ( ! method_exists( seopress_get_service( 'SitemapOption' ), 'isEnabled' ) || '1' !== seopress_get_service( 'SitemapOption' )->isEnabled() ) {
```
Comment (remyperona) [inline inc/ThirdParty/Plugins/SEO/SEOPress.php]: Same here, does it correctly always return an int?

**Review by remyperona — APPROVED** [review]

**Review by vmanthos — APPROVED** [review]
This is working as expected. :+1: 

**NOTES**

The following notices appear in the `debug.log` but are related to SEOPress and _not_ WP Rocket:
```PHP
PHP Deprecated:  Automatic conversion of false to array is deprecated in /shared/httpd/wprocketest/htdocs/wp-content/plugins/wp-seopress/inc/admin/wizard/admin-wizard.php on line 1585
```

With PHP 8.2:
```PHP
PHP Deprecated:  Creation of dynamic property SEOPress\Actions\Admin\Importer\AIO::$tagsAIO is deprecated in /shared/httpd/wprocketest/htdocs/wp-content/plugins/wp-seopress/src/Actions/Admin/Importer/AIO.php on line 12
PHP Deprecated:  Creation of dynamic property SEOPress\Actions\Admin\Importer\RankMath::$tagsRankMath is deprecated in /shared/httpd/wprocketest/htdocs/wp-content/plugins/wp-seopress/src/Actions/Admin/Importer/RankMath.php on line 12
PHP Deprecated:  Creation of dynamic property SEOPress\Actions\Front\Metas\DescriptionMeta::$tagsToStringService is deprecated in /shared/httpd/wprocketest/htdocs/wp-content/plugins/wp-seopress/src/Actions/Front/Metas/DescriptionMeta.php on line 11
PHP Deprecated:  Creation of dynamic property SEOPress\Actions\Front\Metas\TitleMeta::$tagsToStringService is deprecated in /shared/httpd/wprocketest/htdocs/wp-content/plugins/wp-seopress/src/Actions/Front/Metas/TitleMeta.php on line 11
```

**Pull request merged**